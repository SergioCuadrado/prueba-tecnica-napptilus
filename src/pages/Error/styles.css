.error-page {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100%;
    width: 100%;
}

.error-page h2 {
    font-size: 3.2em;
}

.error-page p {
    font-size: 1.2em;
    margin-top: 0;
}

.error-page a {
    text-decoration: none;
    font-size: 1.2em;
    margin-top: 1em;
    background-color: #09f;
    padding: 10px;
    border-radius: 5px;
    color: #fff;
}

.error-page a:hover {
    background-color: #06c;
}