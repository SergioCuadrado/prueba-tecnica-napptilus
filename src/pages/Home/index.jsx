import { Outlet } from 'react-router-dom'
import { Header } from '@/components/Header'
import { ProductsOnCartProvider } from '../../contexts/productsOnCart'

export const Home = () => {
  return (
    <div className="page">
      <ProductsOnCartProvider>
        <Header />
        <Outlet />
      </ProductsOnCartProvider>
    </div>
  )
}
