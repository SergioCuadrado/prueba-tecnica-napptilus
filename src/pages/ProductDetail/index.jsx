import { useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { DetailProduct } from '../../components/DetailProduct'
import { Loader } from '../../components/Loader'
import { useProduct } from '../../hooks/useProduct'

const DETAILS_IN_SPANISH = {
  brand: 'Marca',
  model: 'Modelo',
  price: 'Precio',
  cpu: 'CPU',
  ram: 'RAM',
  os: 'Sistema Operativo',
  screenResolution: 'Resolución de pantalla',
  battery: 'Batería',
  primaryCamera: 'Cámara principal',
  secondaryCamera: 'Cámara secundaria',
  dimensions: 'Dimensiones',
  weight: 'Peso'
}

export const ProductDetail = () => {
  const { id } = useParams()
  const { getDetailProduct, product, error, loading } = useProduct()

  useEffect(() => {
    getDetailProduct({ id })
  }, [])

  const descriptionProducts = product
    ? Object.entries(product?.description).map((desc) => {
      const name = DETAILS_IN_SPANISH[desc[0]]
      const value = desc[1]
      return { name, value }
    }).filter((desc) => {
      return desc.name !== undefined
    })
    : null

  const optionsProduct = product ? product.options : null

  if (error) return <DetailErrorProduct error={error} />
  if (loading) return <Loader />

  return (
    <>
      {product
        ? <DetailProduct product={product} descriptionProducts={descriptionProducts} optionsProduct={optionsProduct} errorValue={error} />
        : null
      }
    </>

  )
}

const DetailErrorProduct = ({ error }) => {
  return (
    <>
      <h2>{error}</h2>
      <p>Por favor, intenta nuevamente más tarde</p>
    </>
  )
}
