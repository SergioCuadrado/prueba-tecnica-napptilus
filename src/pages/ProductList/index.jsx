import { Filter } from '@/components/Filter'
import { Products } from '@/components/Products'
import { useFiltersProducts } from '@/hooks/useFiltersProducts'
import { useProducts } from '@/hooks/useProducts'
import { useEffect } from 'react'

export const ProductList = () => {
  const { filterProducts } = useFiltersProducts()
  const { getProducts, products, loading, error } = useProducts()
  let filteredProducts = []

  useEffect(() => {
    getProducts()
  }, [])

  if (!loading && products?.length > 0) filteredProducts = filterProducts(products)

  if (error) return <h2>{error?.message}</h2>

  return (
    <>
      <Filter />
      <Products products={filteredProducts} loading={loading} />
    </>
  )
}
