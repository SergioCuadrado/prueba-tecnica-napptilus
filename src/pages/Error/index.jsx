import { Link, useRouteError } from 'react-router-dom'

import './styles.css'

export const ErrorPage = () => {
  const error = useRouteError()
  return (
    <section className='error-page'>
        <h2>{error?.status} Error</h2>
        <p>{error?.data}</p>
        <Link to='/'>Volver a la p√°gina inicial</Link>
    </section>
  )
}
