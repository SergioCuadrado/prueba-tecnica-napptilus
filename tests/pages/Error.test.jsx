import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import { ErrorPage } from '@/pages/Error'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteError: () => ({
    status: 404,
    data: 'Not Found'
  })
}))

beforeEach(() => {
  jest.clearAllMocks()
})

describe('Pruebas en la pagina <Error />', () => {
  test('Debe de mostrarse correctamente', () => {
    const { container } = render(
        <MemoryRouter initialEntries={['/prueba']}>
            <ErrorPage />
        </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  test('Si click en el boton de volver a la pagina inicial, debe de redirigir a la pagina inicial', () => {
    const { getByText } = render(
        <MemoryRouter initialEntries={['/prueba']}>
            <ErrorPage />
        </MemoryRouter>
    )
    const link = getByText('Volver a la p√°gina inicial')
    expect(link.getAttribute('href')).toBe('/products')
  })
})
