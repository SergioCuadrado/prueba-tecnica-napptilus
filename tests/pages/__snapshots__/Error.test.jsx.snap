// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la pagina <Error /> Debe de mostrarse correctamente 1`] = `
<div>
  <section
    class="error-page"
  >
    <h2>
      404
       Error
    </h2>
    <p>
      Not Found
    </p>
    <a
      href="/products"
    >
      Volver a la p√°gina inicial
    </a>
  </section>
</div>
`;
